bundles/org.openhab.binding.sony/src/main/resources/ESH-INF/thing/scalar.xml

```diff
@@ -0,0 +1,1720 @@
+		<item-type>Switch</item-type>
+		<label>Power Status</label>
+		<description>The power status of the device</description>
+	</channel-type>
+
+	<channel-type id="scalarsystemwolmode" advanced="true">
+		<item-type>Switch</item-type>
+		<label>WOL Mode</label>
+		<description>The wake on lan (WOL) mode</description>
+	</channel-type>
+
+	<channel-type id="scalarsystemircc">
+		<item-type>String</item-type>
+		<label>IRCC Command</label>
+		<description>The IRCC command to send to the system</description>
+	</channel-type>
+
+	<channel-type id="scalarsystemreboot" advanced="true">
+		<item-type>Switch</item-type>
+		<label>Device Reboot</label>
+		<description>Send 'ON' to reboot the device</description>
+	</channel-type>
+
+	<channel-type id="scalarsystempostalcode" advanced="true">
+		<item-type>String</item-type>
+		<label>Postal Code</label>
+		<description>The postal code of the device</description>
+	</channel-type>
+
+	<channel-type id="scalarsystemdevicename" advanced="true">
+		<item-type>String</item-type>
+		<label>Storage Name</label>
+		<description>The storage name</description>
+		<state readOnly="true" />
+	</channel-type>
+
+	<channel-type id="scalarsystemstorageerror" advanced="true">
+		<item-type>String</item-type>
+		<label>Storage Error</label>
+		<description>Any storage error messages</description>
+		<state readOnly="true" />
+	</channel-type>
+
+	<channel-type id="scalarsystemstoragefilesystem" advanced="true">
+		<item-type>String</item-type>
+		<label>Storage File System</label>
+		<description>The storage file system</description>
+		<state readOnly="true" />
+	</channel-type>
+
+	<channel-type id="scalarsystemstoragefinalizestatus" advanced="true">
+		<item-type>String</item-type>
+		<label>Storage Finalization Status</label>
+		<description>The storage finalization status</description>
+		<state readOnly="true" />
+	</channel-type>
+
+	<channel-type id="scalarsystemstorageformat" advanced="true">
+		<item-type>String</item-type>
+		<label>Storage Format</label>
+		<description>The storage format</description>
+		<state readOnly="true" />
+	</channel-type>
+
+	<channel-type id="scalarsystemstorageformatstatus" advanced="true">
+		<item-type>String</item-type>
+		<label>Storage Format Status</label>
+		<description>The storage format status</description>
+		<state readOnly="true" />
+	</channel-type>
+
+	<channel-type id="scalarsystemstorageformattable" advanced="true">
+		<item-type>String</item-type>
+		<label>Storage Formattable</label>
+		<description>The storage formattable status</description>
+		<state readOnly="true" />
+	</channel-type>
+
+	<channel-type id="scalarsystemstorageformatting" advanced="true">
+		<item-type>String</item-type>
+		<label>Storage Formatting</label>
+		<description>The storage formatting status</description>
+		<state readOnly="true" />
+	</channel-type>
+
+	<channel-type id="scalarsystemstoragefreecapacitymb" advanced="true">
+		<item-type>Number</item-type>
```

Comment:
I think you should seriously consider adding openHab's [Units of Measurement](https://www.openhab.org/docs/concepts/units-of-measurement.html) framework wherever applicable. This allows user interfaces greater flexibility when it comes to rendering data from your binding.
```suggestion
		<item-type>Number:DataAmount</item-type>
```

Reply:
I've never used UOMs - is the ONLY change "number:dataamount" on associated items (both in scalar.xml/code/readmes?  Looking at the doc it seems I should also do a pattern - that's where things get difficult I believe.  The other issue I have is that it won't be complete because sony's api sometimes defines things as numbers and sometimes defines things as strings.  In practice, I think they should be numbers but I didn't want to change from sony's api definition because there may be some reason why they specified a stirng.

Reply:
ack - it looks like I may have to deal with QuantityType as well.  This is a pretty big change and I'm really hesitant to do that

Reply:
This should be changed for all items that can support it. Yes this is a bit change but I highly suggest you do it. UOMs are one of the features that all the maintainers are enforcing in new bindings.

Reply:
@cpmeister 
Started to look at this - but is there support for MB?  Looking at  https://github.com/eclipse/smarthome/tree/master/bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/ there doesn't seem to be support for that

Reply:
You would have to use something like `MetricPrefix.MEGA(SmartHomeUnits.BYTE)`

Reply:
@cpmeister 
ahh - gotcha.  I'll implement this stuff then.  Please note that it will be IMPOSSIBLE to do for general settings however (since those are completely dynamic)

Reply:
@cpmeister 
As it turns out, wasn't that difficult to add (everything seemed to be readonly).  If you could, pay special attention the the changes as I did a lot of copy pasting (and hopefully fixed the correct variable names).

Reply:
@cpmeister 
fyi - as a further note - I did not implement on the items that looked like it could be added but are actual string fields (as per sony spec - such as CN_AUDIOFREQUENCY)